engine/src/execution_graph/logic_controllers/BatchProcessing.h

```diff
@@ -273,12 +298,38 @@ class BindableTableScan : public PhysicalPlan {
 			try {
 				auto batch = input.next();
 
-				if(is_filtered_bindable_scan(expression)) {
-					auto columns = ral::processor::process_filter(batch->toBlazingTableView(), expression, context.get());
-					this->add_to_output_cache(std::move(columns));
-				}
-				else{
-					this->add_to_output_cache(std::move(batch));
+				if (is_filtered_bindable_scan(expression)) {
+					// For large files
+					if (batch->sizeInBytes() > 1073741824) {  // Threshold 1GB
```

Comment:
In this block of code, we should apply the filter and then afterwards see if we want to break it up into smaller batches